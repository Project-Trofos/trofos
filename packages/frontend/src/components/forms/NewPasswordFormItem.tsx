import React from "react"
import { Form, Input } from "antd"

export default function NewPasswordFormItem() : JSX.Element {
    return (
        <>
            <Form.Item
                label="New Password"
                name='newPassword'
                rules={[
                    {
                        required: true,
                        message: 'Please enter a new password'
                    },
                    {
                        min:8,
                        message: 'Password must be atleast 8 characters long'
                    },
                ]}
            >
                <Input type="password"/>
            </Form.Item>
            <Form.Item
                    label="Confirm new Password"
                    name='confirmNewPassword'
                    dependencies={['newPassword']}
                    rules={[
                        {
                            required: true,
                            message: 'Please confirm your password'
                        },
                        ({ getFieldValue }) => ({
                            validator(_, value) {
                                if (!value || getFieldValue("newPassword") === value) {
                                    return Promise.resolve();
                                }
                                return Promise.reject(new Error('The two passwords that you entered do not match!'));
                            }
                        })
                    ]}
                >
                    <Input type="password"/>
            </Form.Item>
        </>
    )
}