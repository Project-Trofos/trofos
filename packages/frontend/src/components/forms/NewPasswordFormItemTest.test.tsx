import { render, screen } from "@testing-library/react";
import React from "react";
import NewPasswordFormItem from "./NewPasswordFormItem";
import FormWrapper from "./__testutils__/FormRenderHelper.test";

describe("NewPasswordFormItem test", () => {
    function renderForm() {
        return render(
            <FormWrapper>
                <NewPasswordFormItem />
            </FormWrapper>
        )
    }

    describe('when rendering', () => {
        it('should have a field for the new password and for its confirmation', () => {
            const { baseElement } = renderForm();
            expect(baseElement).toMatchSnapshot();

            expect(screen.getByLabelText(/^new password/i)).toBeInTheDocument();
            expect(screen.getByLabelText(/confirm new password/i)).toBeInTheDocument();
        })
    })
})