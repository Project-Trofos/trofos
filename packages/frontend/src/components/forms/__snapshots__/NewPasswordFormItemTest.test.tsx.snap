// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewPasswordFormItem test > when rendering > should have a field for the new password and for its confirmation 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="newPassword"
              title="New password"
            >
              New password
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  id="newPassword"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="confirmNewPassword"
              title="Confirm new password"
            >
              Confirm new password
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  id="confirmNewPassword"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default"
        type="submit"
      >
        <span>
          submit
        </span>
      </button>
    </form>
  </div>
</body>
`;
